File: ios-app/MysplitwiseCore/Sources/MysplitwiseCore/Categories.swift
import Foundation

/// Expense categories — port of `src/lib/categories.tsx`. `sfSymbol` names are SF
/// Symbols equivalents chosen to match (not copy) the web's lucide-react icons;
/// a few newer symbol names (iOS 16/17) may need swapping once verified in Xcode.
public struct CategoryInfo {
    public let id: String
    public let name: String
    public let group: String
    public let sfSymbol: String
    public let colorHex: String
}

public enum Categories {
    public static let all: [CategoryInfo] = [
        CategoryInfo(id: "general", name: "General", group: "Uncategorized", sfSymbol: "doc.plaintext", colorHex: "#8E9CA3"),
        CategoryInfo(id: "payment", name: "Payment", group: "Uncategorized", sfSymbol: "banknote", colorHex: "#7C3AED"),

        CategoryInfo(id: "dining", name: "Dining out", group: "Food and drink", sfSymbol: "fork.knife", colorHex: "#FF8A5B"),
        CategoryInfo(id: "groceries", name: "Groceries", group: "Food and drink", sfSymbol: "cart", colorHex: "#7FB069"),
        CategoryInfo(id: "liquor", name: "Liquor", group: "Food and drink", sfSymbol: "wineglass", colorHex: "#B5654C"),
        CategoryInfo(id: "coffee", name: "Coffee", group: "Food and drink", sfSymbol: "cup.and.saucer", colorHex: "#A9744F"),

        CategoryInfo(id: "rent", name: "Rent", group: "Home", sfSymbol: "house", colorHex: "#5B8DC5"),
        CategoryInfo(id: "utilities", name: "Utilities", group: "Home", sfSymbol: "bolt", colorHex: "#F2C14E"),
        CategoryInfo(id: "electricity", name: "Electricity", group: "Home", sfSymbol: "lightbulb", colorHex: "#F2C14E"),
        CategoryInfo(id: "water", name: "Water", group: "Home", sfSymbol: "drop", colorHex: "#5BB6C5"),
        CategoryInfo(id: "internet", name: "Internet", group: "Home", sfSymbol: "wifi", colorHex: "#6C8AE4"),
        CategoryInfo(id: "furniture", name: "Furniture", group: "Home", sfSymbol: "hammer", colorHex: "#9C7B5A"),
        CategoryInfo(id: "household", name: "Household supplies", group: "Home", sfSymbol: "bag", colorHex: "#C58BBB"),

        CategoryInfo(id: "car", name: "Car", group: "Transportation", sfSymbol: "car", colorHex: "#5B7CC5"),
        CategoryInfo(id: "gas", name: "Gas/Fuel", group: "Transportation", sfSymbol: "fuelpump", colorHex: "#E4694A"),
        CategoryInfo(id: "transit", name: "Bus/Train", group: "Transportation", sfSymbol: "bus", colorHex: "#5BA0C5"),

        CategoryInfo(id: "entertainment", name: "Entertainment", group: "Entertainment", sfSymbol: "film", colorHex: "#C566B5"),
        CategoryInfo(id: "tickets", name: "Movies/Tickets", group: "Entertainment", sfSymbol: "ticket", colorHex: "#B05BC5"),
        CategoryInfo(id: "sports", name: "Sports", group: "Entertainment", sfSymbol: "dumbbell", colorHex: "#5BC57F"),

        CategoryInfo(id: "travel", name: "Travel", group: "Life", sfSymbol: "airplane", colorHex: "#5BC5C0"),
        CategoryInfo(id: "shopping", name: "Shopping", group: "Life", sfSymbol: "bag.fill", colorHex: "#E48FB4"),
        CategoryInfo(id: "medical", name: "Medical", group: "Life", sfSymbol: "cross.case", colorHex: "#E45B6E"),
        CategoryInfo(id: "gifts", name: "Gifts", group: "Life", sfSymbol: "gift", colorHex: "#D65BB0"),
        CategoryInfo(id: "education", name: "Education", group: "Life", sfSymbol: "graduationcap", colorHex: "#5B86C5"),
        CategoryInfo(id: "pets", name: "Pets", group: "Life", sfSymbol: "pawprint", colorHex: "#B59A5B"),
        CategoryInfo(id: "kids", name: "Childcare", group: "Life", sfSymbol: "figure.child", colorHex: "#E4A85B"),
    ]

    private static let byId: [String: CategoryInfo] = Dictionary(uniqueKeysWithValues: all.map { ($0.id, $0) })

    public static func get(_ id: String) -> CategoryInfo {
        byId[id] ?? byId["general"]!
    }
}

File: android-app/app/src/main/kotlin/com/mysplitwise/app/logic/Categories.kt
package com.mysplitwise.app.logic

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.*
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * Expense categories — port of `src/lib/categories.tsx`. `icon` is a Material
 * Icons equivalent chosen to match (not copy) the web's lucide-react icons.
 */
data class CategoryInfo(
    val id: String,
    val displayName: String,
    val group: String,
    val icon: ImageVector,
    val colorHex: String,
)

object Categories {
    val all: List<CategoryInfo> by lazy {
        listOf(
            CategoryInfo("general", "General", "Uncategorized", Icons.Outlined.Receipt, "#8E9CA3"),
            CategoryInfo("payment", "Payment", "Uncategorized", Icons.Outlined.Payments, "#7C3AED"),

            CategoryInfo("dining", "Dining out", "Food and drink", Icons.Outlined.Restaurant, "#FF8A5B"),
            CategoryInfo("groceries", "Groceries", "Food and drink", Icons.Outlined.ShoppingCart, "#7FB069"),
            CategoryInfo("liquor", "Liquor", "Food and drink", Icons.Outlined.LocalBar, "#B5654C"),
            CategoryInfo("coffee", "Coffee", "Food and drink", Icons.Outlined.LocalCafe, "#A9744F"),

            CategoryInfo("rent", "Rent", "Home", Icons.Outlined.Home, "#5B8DC5"),
            CategoryInfo("utilities", "Utilities", "Home", Icons.Outlined.Bolt, "#F2C14E"),
            CategoryInfo("electricity", "Electricity", "Home", Icons.Outlined.Lightbulb, "#F2C14E"),
            CategoryInfo("water", "Water", "Home", Icons.Outlined.WaterDrop, "#5BB6C5"),
            CategoryInfo("internet", "Internet", "Home", Icons.Outlined.Wifi, "#6C8AE4"),
            CategoryInfo("furniture", "Furniture", "Home", Icons.Outlined.Chair, "#9C7B5A"),
            CategoryInfo("household", "Household supplies", "Home", Icons.Outlined.ShoppingBag, "#C58BBB"),

            CategoryInfo("car", "Car", "Transportation", Icons.Outlined.DirectionsCar, "#5B7CC5"),
            CategoryInfo("gas", "Gas/Fuel", "Transportation", Icons.Outlined.LocalGasStation, "#E4694A"),
            CategoryInfo("transit", "Bus/Train", "Transportation", Icons.Outlined.DirectionsBus, "#5BA0C5"),

            CategoryInfo("entertainment", "Entertainment", "Entertainment", Icons.Outlined.Theaters, "#C566B5"),
            CategoryInfo("tickets", "Movies/Tickets", "Entertainment", Icons.Outlined.ConfirmationNumber, "#B05BC5"),
            CategoryInfo("sports", "Sports", "Entertainment", Icons.Outlined.FitnessCenter, "#5BC57F"),

            CategoryInfo("travel", "Travel", "Life", Icons.Outlined.Flight, "#5BC5C0"),
            CategoryInfo("shopping", "Shopping", "Life", Icons.Outlined.ShoppingBag, "#E48FB4"),
            CategoryInfo("medical", "Medical", "Life", Icons.Outlined.MedicalServices, "#E45B6E"),
            CategoryInfo("gifts", "Gifts", "Life", Icons.Outlined.CardGiftcard, "#D65BB0"),
            CategoryInfo("education", "Education", "Life", Icons.Outlined.School, "#5B86C5"),
            CategoryInfo("pets", "Pets", "Life", Icons.Outlined.Pets, "#B59A5B"),
            CategoryInfo("kids", "Childcare", "Life", Icons.Outlined.ChildCare, "#E4A85B"),
        )
    }

    private val byId: Map<String, CategoryInfo> by lazy { all.associateBy { it.id } }

    fun get(id: String): CategoryInfo = byId[id] ?: byId.getValue("general")
}
